persistence/jdbc/src/main/java/org/infinispan/persistence/jdbc/stores/AbstractJdbcStore.java

```diff
@@ -0,0 +1,204 @@
+      }
+   }
+
+   @Override
+   public void clear() {
+      Connection conn = null;
+      PreparedStatement ps = null;
+      try {
+         String sql = tableManager.getDeleteAllRowsSql();
+         conn = connectionFactory.getConnection();
+         ps = conn.prepareStatement(sql);
+         int result = ps.executeUpdate();
+         if (log.isTraceEnabled()) {
+            log.tracef("Successfully removed %d rows.", result);
+         }
+      } catch (SQLException ex) {
+         log.failedClearingJdbcCacheStore(ex);
+         throw new PersistenceException("Failed clearing cache store", ex);
+      } finally {
+         JdbcUtil.safeClose(ps);
+         connectionFactory.releaseConnection(conn);
+      }
+   }
+
+   @Override
+   public void commit(Transaction tx) {
+      Connection connection;
+      try {
+         connection = getTxConnection(tx);
+         connection.commit();
+      } catch (SQLException e) {
+         log.sqlFailureTxCommit(e);
+         throw new PersistenceException(String.format("Error during commit of JDBC transaction (%s)", tx), e);
+      } finally {
+         destroyTxConnection(tx);
+      }
+   }
+
+   @Override
+   public void rollback(Transaction tx) {
+      Connection connection;
+      try {
+         connection = getTxConnection(tx);
+         connection.rollback();
+      } catch (SQLException e) {
+         log.sqlFailureTxRollback(e);
+         throw new PersistenceException(String.format("Error during rollback of JDBC transaction (%s)", tx), e);
+      } finally {
+         destroyTxConnection(tx);
+      }
+   }
+
+   protected Connection getTxConnection(Transaction tx) {
+      Connection connection = transactionConnectionMap.get(tx);
+      if (connection == null) {
+         connection = connectionFactory.getConnection();
+         transactionConnectionMap.put(tx, connection);
+      }
+      return connection;
+   }
+
+   protected void destroyTxConnection(Transaction tx) {
+      Connection connection = transactionConnectionMap.remove(tx);
+      if (connection != null)
+         connectionFactory.releaseConnection(connection);
+   }
+
+   /**
+    * Keeps a reference to the connection factory for further use. Also initializes the {@link
+    * TableManager} that needs connections. This method should be called when you don't
+    * want the store to manage the connection factory, perhaps because it is using an shared connection factory: see
+    * {@link org.infinispan.persistence.jdbc.stores.mixed.JdbcMixedStore} for such an example of this.
+    */
+   public void initializeConnectionFactory(ConnectionFactory connectionFactory) throws PersistenceException {
+      this.connectionFactory = connectionFactory;
+      tableManager = getTableManager();
+      tableManager.setCacheName(cacheName);
+      tableManager.start();
+   }
+
+   public ConnectionFactory getConnectionFactory() {
+      return connectionFactory;
+   }
+
+   protected void waitForFutureToComplete(Future<Void> future) {
```

Comment:
@danberindei I agree, there are a few issues with the implementations of purge and process in the String based store, I simply moved the existing code and reduced some repetition with `waitForFutureToComplete`. My plan is to address these issues in a future PR as part of [1][2] as this PR already contains significant changes.

[1] https://issues.jboss.org/browse/ISPN-6843
[2]https://issues.jboss.org/browse/ISPN-6712